app/src/main/java/org/p2p/wallet/send/model/SendFee.kt

```diff
@@ -1,54 +1,138 @@
 package org.p2p.wallet.send.model
 
 import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.send.model.FeePayerSelectionStrategy.CORRECT_AMOUNT
+import org.p2p.wallet.send.model.FeePayerState.ReduceInputAmount
+import org.p2p.wallet.send.model.FeePayerState.SwitchToSol
+import org.p2p.wallet.send.model.FeePayerState.UpdateFeePayer
+import org.p2p.wallet.utils.Constants.SOL_SYMBOL
+import org.p2p.wallet.utils.fromLamports
+import org.p2p.wallet.utils.isLessThan
+import org.p2p.wallet.utils.isMoreThan
+import org.p2p.wallet.utils.scaleMedium
 import org.p2p.wallet.utils.toLamports
 import org.p2p.wallet.utils.toUsd
 import java.math.BigDecimal
 import java.math.BigInteger
 
-sealed class SendFee(
-    open val fee: BigDecimal,
-    open val feePayerToken: Token.Active,
-    open val sourceTokenSymbol: String
-) {
-
-    val feeLamports: BigInteger
-        get() = fee.toLamports(feePayerToken.decimals)
-
-    val feeTotalLamports: BigInteger
-        get() = feePayerToken.total.toLamports(feePayerToken.decimals)
-
-    val feeUsd: BigDecimal?
-        get() = fee.toUsd(feePayerToken)
+sealed interface SendFee {
 
+    val feePayerToken: Token.Active
     val formattedFee: String
-        get() = "${fee.toPlainString()} ${feePayerToken.tokenSymbol}"
+    val sourceTokenSymbol: String
+    val feeUsd: BigDecimal?
+    val feeDecimals: BigDecimal
 
-    val feePayerSymbol: String
-        get() = feePayerToken.tokenSymbol
+    fun isEnoughToCoverExpenses(sourceTokenTotal: BigInteger, inputAmount: BigInteger): Boolean
 
-    data class RenBtcFee(
-        override val fee: BigDecimal,
+    class RenBtcFee(
         override val feePayerToken: Token.Active,
+        private val feeLamports: BigInteger
```

Comment:
Is it definitely lamports value if token is not SOL? Or it just 0.000000001 from the token unit?

Reply:
didn't understand your question

Reply:
I could be wrong but I guess that lamports definition could be applied to SOL only, but we use this for all tokens, meaning just that fractional value